packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -760,27 +762,43 @@ export class Function extends FunctionBase {
     return this._logGroup;
   }
 
+  public checkEdgeCompatibility(): void {
+    // Check env vars
+    const envEntries = Object.entries(this.environment);
+    for (const [key, config] of envEntries) {
+      if (config.removeInEdge) {
+        delete this.environment[key];
+        this.node.addInfo(`Removed ${key} environment variable for Lambda@Edge compatibility`);
+      }
+    }
+    const envKeys = Object.keys(this.environment);
+    if (envKeys.length !== 0) {
+      throw new Error(`The function ${this.node.path} is not compatible for Lambda@Edge because it contains environment variables that are not marked for removal when used for Lambda@Edge: ${envKeys}. Use \`addEnvironment()\` to allow edge removal.`);
```

Comment:
Line wrap this to 120/150.

Reply:
Minor language tweak.

```suggestion
      throw new Error(`The function ${this.node.path} contains environment variables [${envKeys}] and is not compatible with Lambda@Edge. Environment variables can be marked for removal when used in Lambda@Edge by setting the 'removeInEdge' property in the 'addEnvironment()' API.`);
```